packages/peregrine/lib/talons/Checkout/useFlow.js

```diff
@@ -62,6 +71,12 @@ export const useFlow = props => {
         setStep('cart');
     }, [setStep]);
 
+    useEffect(() => {
+        if (countriesData) {
+            setCountries(countriesData.countries);
```

Comment:
Unfortunately we still do have to persist countries data in redux because the `formatAddress` function needs `countries` available to format with.

Reply:
Actually ... I might be able to pass `countries` down at the time `submitPaymentMethodAndBillingAddress` is called 👍

Reply:
Yea I don't think you really need to do that anymore -- it's basically local state now. You can pass it to the shipping form and the billing/payment form and they can use some utility function there.

Reply:
Do you think `flow` is the best place for this? Or would the `form` component be the more appropriate? I wasn't sure how much prop threading we wanted to encourage.

Reply:
Redux now no longer needs to track `countries` https://github.com/magento/pwa-studio/pull/1993/commits/68cd2b8ac59650c6a8b8a677a46d355adf1d1c0b and https://github.com/magento/pwa-studio/pull/1993/commits/0d3572a06546452552dd9e4ed6fff36b5857244a.

> Do you think flow is the best place for this? Or would the form component be the more appropriate? I wasn't sure how much prop threading we wanted to encourage.

I had just kept it in the same place it was, but yeah it can definitely be moved down a level. Working on that now.

Reply:
I was able to move it down to `EditableForm` 👍 

https://github.com/magento/pwa-studio/pull/1993/commits/daa0cb90a220420fa8d4ea1c5d5e9e3fe14a84ca.

The countries data isn't populated until the user starts to edit the checkout form:

![Screen Shot 2019-11-22 at 8 30 16 AM](https://user-images.githubusercontent.com/13182778/69434056-7d35e080-0d02-11ea-97b4-05f9f246c424.png)

Less prop threading but also may open us up to race conditions if the GraphQL query is slow? We need the countries data by the time that "Use Address" button is clicked (for validation).

Reply:
Okay I just simulated it with a slow 3G connection: our validators are smart and early out. The first validation is that the "State" field is required, which doesn't need the `countries` data to do its job.

I'm just thinking if there's an auto-filled form or something, we may be vulnerable to a case where the user clicks "Use Address" before we have `countries` data.

Reply:
Ok, the countries data is now fetched when the Checkout Form is shown (when this "Checkout" button is clicked): 

![Screen Shot 2019-11-22 at 8 47 46 AM](https://user-images.githubusercontent.com/13182778/69435284-def74a00-0d04-11ea-954f-8e84d6827472.png)

It only threads down to `EditableForm` > `AddressForm` and `EditableForm` > `PaymentsForm`, and the GraphQL call should return by the time those forms need it for validation.